apps/nextjs-website/src/app/auth/confirmation/page.tsx

```diff
@@ -1,56 +1,90 @@
 'use client';
-import { Box, CircularProgress, Stack } from '@mui/material';
+import { Button } from '@mui/material';
 import PageNotFound from '@/app/not-found';
 import { Auth } from 'aws-amplify';
 import { useSearchParams, useRouter } from 'next/navigation';
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
+import Spinner from '@/components/atoms/Spinner/Spinner';
+import FileErrorIcon from '@/components/atoms/FileErrorIcon/FileErrorIcon';
+import { useTranslations } from 'next-intl';
+import PageBackgroundWrapper from '@/components/atoms/PageBackgroundWrapper/PageBackgroundWrapper';
+import SingleCard from '@/components/atoms/SingleCard/SingleCard';
+import { isProduction } from '@/config';
+
+enum State {
+  loading = 'loading',
+  resendCode = 'resendCode',
+  error = 'error',
+}
 
 const Confirmation = () => {
   const searchParams = useSearchParams();
   const router = useRouter();
   const username = searchParams.get('username');
   const code = searchParams.get('code');
+  const t = useTranslations('auth');
 
-  const [renderNotFound, setRenderNotFound] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
+  const [state, setState] = useState<State>(State.loading);
 
   useEffect(() => {
     if (username && code) {
+      console.log(username, code);
       Auth.confirmSignUp(username, code)
         .then(() => {
-          setRenderNotFound(false);
           // eslint-disable-next-line functional/immutable-data
           router.push('/auth/account-activated');
         })
-        .catch(() => {
-          setRenderNotFound(true);
+        .catch((error) => {
+          // TODO: remove console warn and handle errors: [CodeMismatchException, ExpiredCodeException, InternalErrorException, LimitExceededException]
```

Comment:
this is TODO is relative to this issue https://pagopa.atlassian.net/browse/DEV-1179